docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,158 @@
+write a ``good_file`` which has less than or equal to 10 lines and a
+``bad_file`` which has more than 10 lines and feed them to
+``verify_local_bear`` as input along with your bear (MyBear in this case) and
+a few additional arguments.
+
+.. note::
+
+    ``good_file`` and ``bad_file`` are sequences just like ``file``. A ``file``
+    is a sequence of an input file.
+
+**EXAMPLE 2** using ``LocalBearTestHelper.check_validity``
+
+.. code::
+
+    from queue import Queue
+    from bears.python.MyBear import MyBear
+
+    from tests.LocalBearTestHelper import LocalBearTestHelper
+    from coalib.settings.Section import Section
+    from coalib.settings.Setting import Setting
+
+
+    class MyBearTest(LocalBearTestHelper):
+
+        def setUp(self):
+            self.section = Section('name')
+            self.section.append(Setting('max_number_of_lines', '10'))
+            self.uut = MyBear(self.section, Queue())
+
+        def test_valid(self):
+            self.check_validity(self.uut, ["import os"])
+
+        def test_invalid(self):
+            self.check_validity(self.uut, file, valid=False)
+
+.. note::
+
+    ``file`` contains more than 10 lines.
+
+``check_validity`` asserts if your bear yields any results for a particular
+check with a list of strings. First a *Section* and your Bear
+(in this case MyBear) is ``setUp``. Now your *Section* consists on default
+*Settings*. You can append any *Setting* depending on your test. Validate a
+check by passing your bear, lines to check as parameters
+(pass few others parameters if necessary) to ``check_validity``. The method
+``self.check_validity(self.utt, ["import os"]`` asserts if your bear
+``self.utt`` yields a result if a list of strings ``["import os"]`` is passed.
+
+**EXAMPLE 3** using ``LocalBearTestHelper.check_results``
+
+.. code::
+
+    from queue import Queue
+
+    from bears.somelanguage.MyBear import MyBear
+    from tests.LocalBearTestHelper import LocalBearTestHelper
+    from coalib.results.Result import Result
+    from coalib.settings.Section import Section
+
+
+    class MyBearTest(LocalBearTestHelper):
+
+        def setUp(self):
+            self.uut = MyBear(Section('name'), Queue())
+
+        def test_run(self):
+            self.check_results(
+                self.uut,
+                file,
+                Result.from_values("MyBear",
+                                   "Too many lines")
+                )
+
+``check_results`` asserts if your bear results match the actual
+results on execution on CLI. Just like the above example, we need to ``setUp``
+a *Section* and your Bear with some *Settings*. ``check_results`` validates
+your results by giving your local bear, lines to check and expected results
+as input. ``check_results`` asserts if your bear's results on checking the
+``file`` matches with the ``Results.from_values(...)``.
+
+A Final Note
+------------
+
+LocalBearTestHelper is written to ease off testing for bears. Make sure that
```

Comment:
``LocalBearTestHelper`` might be used in italic or with double backticks, i'm not so sure if it should be since you haven't used it before, probably it shouldn't. if noone else comments regarding this, ignore this.